### ISSUE TITLE
Allow users to exclude one or more facet values from search results

### ISSUE THREAD

**Issue by andrewjbtw** [sul-dlss/argo#4222]
This is an update to https://github.com/sul-dlss/argo/issues/921

Users have requested the ability to exclude sets of search results based on facet values. The idea would be to:

- Go to the facet list
- Select a value from the facet to be excluded from the search results

Example limits:

- show me all results where the access rights are NOT dark
- show me all items where the processing status is NOT registered

See screenshot and link for the design, which includes an outline of how the interaction would work as a sequence of steps:

![Screenshot 2023-10-09 at 14-16-06 exclude](https://github.com/sul-dlss/argo/assets/8975583/e5d14726-71d6-4e1e-ba47-d348aa4025f3)
https://projects.invisionapp.com/share/B813BT2Q9MRH#/screens/474829031_exclude

**Comment by ndushay** [comment]
We may be able to leverage Columbia's work.  I think this may be part of their key:  https://github.com/cul/clio-spectrum/blob/main/app/controllers/catalog_controller.rb#L260

We can do this in the URL (note the minus sign before the facet field name:

https://argo-qa.stanford.edu/catalog?f[-collection_title_ssim][]=Arrowroot+Curry+Leaves
vs
https://argo-qa.stanford.edu/catalog?f[collection_title_ssim][]=Arrowroot+Curry+Leaves

**Comment by ndushay** [comment]
From JCoyne:

> I think you're looking for https://github.com/cul/clio-spectrum/commit/2d04055daa1bc2dfae05802e2671ba40e80a607c - " But that was written 10 years ago so it’ll need to be forward ported to whatever version of BL you’re on now"

**Comment by andrewjbtw** [comment]
I added a top 5 list of facets to the original description if it turns out it's more viable to enable this on a per-facet basis.

**Comment by mjgiarlo** [comment]
@andrewjbtw @astridu I've started spiking on this in Blacklight, and I have an approach that has the baseline functionality working (for a very small sample size). I need technical feedback to make it scalable and palatable to the community, but once I get that, I'll want to turn to implementing the UI design so that I no longer need to manually hack the URL to drive this functionality. The UI design above looks great! I believe we need one additional design, namely for the little constraint buttons that render as you add more facets. So, for example:

![Screenshot from 2023-10-12 16-36-54](https://github.com/sul-dlss/argo/assets/131982/c0277a40-e45f-4f1a-b726-88c7a6dd97b9)

This example shows what the constraint buttons show when faceting for all docs that have English as a language and German as a language. How would this look when showing all docs that have English as a language **EXCLUDING** docs with German as a language?

**Comment by astridu** [comment]
Yes, I can do that. Could I get it to you by 10/27, or do you need it sooner?

**Comment by mjgiarlo** [comment]
@astridu No rush. 😄 Thank you!

**Comment by astridu** [comment]
Here is an option. We would probably want to test it before it went out, but if you're looking for a way to prototype it, this could work. 
![image](https://github.com/sul-dlss/argo/assets/42216982/806f0d1f-5e9f-4ae8-9c12-42c7822400d9)

**Comment by justinlittman** [comment]
Backlogging until Blacklight update.
